katcp/core.py

```diff
@@ -299,7 +299,17 @@ def format_argument(self, arg):
         elif isinstance(arg, bool):
             return str(int(arg))
         else:
-            return str(arg)
+            try:
+                return str(arg)
+            except UnicodeEncodeError:
+                # unicode characters will break the str cast, so
+                # try to encode to ascii and replace the offending characters
+                # with a '?' character
+                logger.error("Error casting message argument to str! "
+                             "Trying to encode argument to ascii.")
+                if not isinstance(arg, unicode):
+                    arg = arg.decode('utf-8')
+                return arg.encode('ascii', 'replace')
```

Comment:
Since there are no easy way to identify the message in the error log (it seems) I suggest to replace the arg with '' as it is easier to notice than finding one ? in a long description.